aten/src/ATen/native/mps/operations/BinaryKernel.mm

```diff
@@ -323,7 +323,7 @@ static void binary_mps_impl(TensorIteratorBase& iter, const std::string func_nam
 }
 
 void complex_mul_out(const Tensor& input, const Tensor& other, const Tensor& output) {
-  TORCH_INTERNAL_ASSERT(c10::isComplexType(input.scalar_type()) && c10::isComplexType(other.scalar_type()));
+  TORCH_INTERNAL_ASSERT(c10::isComplexType(input.scalar_type()) || c10::isComplexType(other.scalar_type()));
```

Comment:
I think issue is a bit more involved, because first of all we should not be using this code at all on MacOS 14 (this was a intermediate solution for MacOS 12/13, but should not be used by default)

If you don't mind, I plan to submit a PR later today to address that and few other issues together